src/engraving/libmscore/slur.cpp

```diff
@@ -366,8 +366,44 @@ void SlurSegment::adjustEndpoints()
     }
 }
 
+Shape SlurSegment::getSegmentShape(Segment* seg, ChordRest* startCR, ChordRest* endCR)
+{
+    staff_idx_t startStaffIdx = startCR->staffIdx();
+    Shape segShape = seg->staffShape(startStaffIdx).translated(seg->pos() + seg->measure()->pos());
+    // If cross-staff, also add the shape of second staff
+    if (slur()->isCrossStaff() && seg != startCR->segment()) {
+        staff_idx_t endStaffIdx = (endCR->staffIdx() != startStaffIdx) ? endCR->staffIdx() : endCR->vStaffIdx();
+        SysStaff* startStaff = system()->staves().at(startStaffIdx);
+        SysStaff* endStaff = system()->staves().at(endStaffIdx);
+        double dist = endStaff->y() - startStaff->y();
+        Shape secondStaffShape;
+        secondStaffShape.add(seg->staffShape(endStaffIdx).translated(seg->pos() + seg->measure()->pos()));
+        secondStaffShape.translate(PointF(0.0, dist));
+        segShape.add(secondStaffShape);
+    }
+    // Remove items that the slur shouldn't try to avoid
+    mu::remove_if(segShape, [startCR](ShapeElement& shapeEl){
+        if (!shapeEl.toItem) {
+            return false;
+        }
+        // Lyrics and fingering
+        const EngravingItem* item = shapeEl.toItem;
+        if (item->isLyrics() || item->isFingering()) {
+            return true;
+        }
+        // Items belonging to its own start chord
+        EngravingItem* parentItem = item->parentItem();
+        if (parentItem && parentItem->isChord() && toChord(parentItem) == startCR) {
```

Comment:
Do you specifically want to check for Chords rather than ChordRests here? 

If no, then this could be just `if (item->parentItem() == startCR)`. Non-nullness of `item->parentItem()` is then guaranteed because `startCR` is not null; and `toChord` returns exactly the same pointer as you feed it, only C++ interprets it as a pointer to a different type, but it will be the same pointer with the same address. And if the address is the same, you are already sure that it is _exactly_ the same object in memory and that it _must_ be a ChordRest because `startCR` is a ChordRest.

If yes, then `toChord` is still not necessary, but won't do much harm either, since it does (almost) nothing at runtime, but rather at compile time.

Reply:
In the rare occurrence that a slur starts on a rest, the rest normally doesn't carry any item (flags are especially annoying) that may interfere with the slur position, hence why I was excluding it. Though I think it is indeed easier (and more general) if I just compare the ChordRest pointer